cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -90,12 +90,19 @@ public Location call() throws Exception {
       Iterable<Location> streamLocations = StreamUtils.listAllStreams(streamBaseLocation);
 
       for (final Location streamLocation : streamLocations) {
-        StreamId streamId = namespace.getNamespaceId().stream(StreamUtils.getStreamNameFromLocation(streamLocation));
+        final StreamId streamId = namespace.getNamespaceId().stream(StreamUtils
+                                                                      .getStreamNameFromLocation(streamLocation));
         final AtomicLong ttl = new AtomicLong(0);
-        if (isStreamExists(streamId)) {
-          ttl.set(streamAdmin.getConfig(streamId).getTTL());
-        }
-        clean(streamLocation, ttl.get(), System.currentTimeMillis());
+        impersonator.doAs(streamId, new Callable<Void>() {
+          @Override
+          public Void call() throws Exception {
+            if (isStreamExists(streamId)) {
+              ttl.set(streamAdmin.getConfig(streamId).getTTL());
+            }
+            clean(streamLocation, ttl.get(), System.currentTimeMillis());
```

Comment:
@anwar6953 has comment about this https://github.com/caskdata/cdap/pull/7698#discussion_r99262107